edx_exams/apps/core/tests/test_api.py

```diff
@@ -289,6 +290,74 @@ def test_with_no_attempt(self):
         self.assertIsNone(get_attempt_by_id(111111111))
 
 
+@ddt.ddt
+class TestGetLatestAttemptForUser(ExamsAPITestCase):
+    """
+    Test for the API utility function `get_latest_attempt_for_user`
+    """
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = '11111111'
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+    def create_mock_exam_attempt(self, user, attempt_number, status, start_time):
```

Comment:
sorry for not catching this before, but the attempt_number can always be 1 for these attempts (or should default to 1). It's an externally used value, and is always set to 1 on exam attempt creation